pkg/util/mon/bytes_usage.go

```diff
@@ -551,6 +553,30 @@ func (b *BoundAccount) Init(ctx context.Context, mon *BytesMonitor) {
 	b.mon = mon
 }
 
+// Reserve requests an allocation of some amount from the monitor just like Grow
+// but does not mark it as used immediately, instead keeping it in the local
+// reservation by use by future Grow() calls, and configuring the account to
+// consider that amount "earmarked" for this account, meaning that that Shrink()
+// calls will not release it back to the parent monitor.
+//
+// If Mu is set, it is safe for use by concurrent goroutines.
+func (b *BoundAccount) Reserve(ctx context.Context, x int64) error {
+	if b == nil {
+		return nil
+	}
+	if b.Mu != nil {
+		b.Mu.Lock()
+		defer b.Mu.Unlock()
+	}
+	minExtra := b.mon.roundSize(x)
+	if err := b.mon.reserveBytes(ctx, minExtra); err != nil {
+		return err
+	}
+	b.reserved += minExtra
+	b.earmark = x
```

Comment:
is this intentionally b.earmark = x instead of b.earmark += x? Maybe I'm missing something here.

Reply:
/shrug. 

Not intentionally. I don't think it'd obviously wrong either way -- calling reserve with a different value could reasonably be expected to change the reserved amount, or to add to it, so I don't feel strongly either way. I guess add to it makes more sense if we're unconditionally growing actually bytes reserved from the parent?

Reply:
> I guess add to it makes more sense if we're unconditionally growing actually bytes reserved from the parent?

Yup, this makes sense to me.

Reply:
Done.